nixos/tests/airsonic.nix

```diff
@@ -0,0 +1,29 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "airsonic";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ sumnerevans ];
+  };
+
+  machine =
+    { pkgs, ... }:
+    {
+      services.airsonic.enable = true;
+    };
+
+  testScript = ''
+    def airsonic_is_up(_) -> bool:
+        return machine.succeed("curl --fail http://localhost:4040/login")
+
+
+    machine.start()
+    machine.wait_for_unit("airsonic.service")
+    machine.wait_for_open_port(4040)
+
+    with machine.nested("Waiting for UI to work"):
+        retry(airsonic_is_up)
+  '';
+
+  # Airsonic is a Java application, and unfortunately requires a significant
+  # amount of memory.
+  virtualisation.memorySize = 1024;
```

Comment:
This, somewhat counter-intuitively, needs to be in the machine description.

You probably also want to set services.airsonic.maxMemory = 800;

Then the test passes for me in about a minute and a half without too much GC thrashing.

Reply:
Oh, that makes sense. Fixed. 🤞 that the hydra build works.